### ISSUE TITLE
long additions

### ISSUE THREAD

**Issue by mmwwah** [scottkleinman/aeme#81]
The day has come, when I need to deal with a lengthy (105 lines) addition, which is 5pp. later in the codex but connected with a signe de renvoi.  The `<add>` and `<addition>` tags seem to only cover short passages.  I assume that I will want to use `<addSpan>` and `<anchor>` here, right?  

Is there some algorithm or heuristic by which we give an xml id to an anchor?  And what connects it with the `<addSpan>`, other than it being the next anchor that follows it?  

Am I right in thinking that I should use `<@rend>` to indicate that which folio the addition is physically located?

**Comment by scottkleinman** [comment]
Here is some markup of a classic poem. Lines 2-3 are an added span, and there is a *signe de renvoi* (here assumed to be a line) connecting line 2 to line 1.

```xml
 <!-- Running text -->
 <l xml:id="line1">Mistress Mary, Quite contrary,</l>

 <!-- Added lines -->
 <addSpan n="added lines" spanTo="#l3" place="5r#marginRight"/>
   <l xml:id="line2">
     <metamark function="line-connector" rend="line" target="line1"/>
     How does your garden grow?
   </l>
   <l xml:id="line3">With Silver Bells, And Cockle Shells,</l>
 <anchor xml:id="l3"/>
 <note type="enc" resp="YOUR_XML:ID">
    Lines 2-3 are added on folio 5r with a signe de renvoi drawn 
    connecting them to line 1.
  </note>

 <!-- Continue running text -->
 <l xml:id="line4">Sing cuckolds all in a row.</l>
```

I think this short example illustrates that `<anchor>` functions like an end tag for `<addSpan>` and that assigning it an `@xml:id` is simple. Here the value is simply an abbreviation for "line 3". You could also choose "line 4" for this purpose; we don't currently have a standard means of referring to anchors. Like all instances of `@xml:id`, the only requirement is that the value be unique in the file.

My sense is that `@place` should be used to indicate the placement of the addition. I base this on the face that the TEI guidelines give "overleaf" as a suggested value. But this seems insufficient to me, as it does not indicate where on the verso the addition appears (in the primary text, one of the margins?). Additionally, we would want to refer to a folio number. `@place="5r#marginRight"` in the example above could be parsed in order to place the addition in the right margin of folio 5r. I'm not sure how this could possibly be rendered by the stylesheet, so I would recommend that the addition be displayed between lines 1 and 2 (in the example above) in both the critical and diplomatic views. This is why I think the note after the anchor in the example above is useful. The valuable information accessible, even if it is in a more traditional form of annotation.

We'll have to tackle this again when we do the facsimile view.

**Comment by mmwwah** [comment]
Great -- thank you!

**Comment by dorothyk98** [comment]
Just out of curiosity, what does this mean in terms of how this will look from Facsimile layer (where the extra large addition is somewhere else) to the critical layer (where we will want to read it in the order it's supposed to be)...

**Comment by mmwwah** [comment]
I assume the FacsV will look like the MS -- in other words, invite the same error that Holt & White made, in not recognizing the insertion for what it was.  I can't imagine a FacsV in which it wouldn't.

**Comment by scottkleinman** [comment]
Meg never mentioned what the *signe de renvoi* is, but she is essentially right that what you see in the facsimile view is what's on the individual page. I'm still working on getting an image server up that will allow us to deploy the Mirador viewer. This will allow viewing the two non-consecutive pages side by side. Of course, you have to know to do that. The metamark target provides the necessary encoding, but implementing it will be a devil...

**Comment by mmwwah** [comment]
He uses his little eyeball signe de renvoi:
![screen shot 2015-12-11 at 13 45 04](https://cloud.githubusercontent.com/assets/7809545/11752413/7a596d66-a00d-11e5-9080-630b7af495dd.png)

Five folia later, you see it again, and if you have your thinking cap on (or have read the literature), you recognize it as an insertion rather than a whole new bizarre preface.

**Comment by skgoetz** [comment]
If it isn't possible to render the invoked text segment (I would suggest that it's not ideal to render the invoked segment at point of invocation), generating a link from that bit of TEI encoding ought certainly to be doable. 
-- 
Sent via phone.

**Comment by mmwwah** [comment]
You're talking FacsV here, right, Sharon?

**Comment by skgoetz** [comment]
I take back the desirability part: was confused about which view was being discussed. 
-- 
Sent via phone.

**Comment by scottkleinman** [comment]
Yes, in the facsimile view, this would essentially link to a free-floating text segment. We'd need to somehow script the surrounding context. For instance, click the metamark and the page containing the targeted text appears alongside the current page. It probably wouldn't be hard to add a function to Mirador to do this, but you would need to generate a special link to call the function. Or maybe not. Maybe you'd just have the function locate the target, check whether it's on the same page, and act accordingly. Regardless, it requires an extra bit of UI development.

**Comment by dorothyk98** [comment]
Also, that eye is so visually wow. So I assume in the critical edition view, there would be some eye, but then the added text would then be there for people to read. Thoughts Meg on what this means about visual vs. verbal performance and about reading the text?
